tests/tracked_token_accounts.test.ts

```diff
@@ -0,0 +1,493 @@
+        },
+        {
+          address: "account4",
+          owner_address: "owner4",
+          token_id: tokenId,
+          first_transaction_date: new Date("2022-06-03"),
+        },
+      ]);
+
+    // balances for 6/2 and 6/3 should be added
+    const updatedTokenAccountBalances = await knex<TrackedTokenAccountBalance>(
+      "tracked_token_account_balances"
+    ).select("*");
+    chai
+      .expect(updatedTokenAccountBalances)
+      .excluding("id")
+      .to.eql([
+        {
+          tracked_token_account_id: updatedTokenAccounts.find(
+            (account) => account.address === "account1"
+          ).id,
+          datetime: new Date("2022-06-01"),
+          approximate_minimum_balance: "10",
+        },
+        {
+          tracked_token_account_id: updatedTokenAccounts.find(
+            (account) => account.address === "account2"
+          ).id,
+          datetime: new Date("2022-06-01"),
+          approximate_minimum_balance: "20",
+        },
+        // make sure 6/2 balances include the unchanged account2 from 6/1
+        {
+          tracked_token_account_id: updatedTokenAccounts.find(
+            (account) => account.address === "account1"
+          ).id,
+          datetime: new Date("2022-06-02"),
+          approximate_minimum_balance: "0",
+        },
+        {
+          tracked_token_account_id: updatedTokenAccounts.find(
+            (account) => account.address === "account2"
+          ).id,
+          datetime: new Date("2022-06-02"),
+          approximate_minimum_balance: "20",
+        },
+        {
+          tracked_token_account_id: updatedTokenAccounts.find(
+            (account) => account.address === "account3"
+          ).id,
+          datetime: new Date("2022-06-02"),
+          approximate_minimum_balance: "30",
+        },
+        // make sure the 6/3 balances include all previous days too
+        {
+          tracked_token_account_id: updatedTokenAccounts.find(
+            (account) => account.address === "account1"
+          ).id,
+          datetime: new Date("2022-06-03"),
+          approximate_minimum_balance: "0",
+        },
+        {
+          tracked_token_account_id: updatedTokenAccounts.find(
+            (account) => account.address === "account2"
+          ).id,
+          datetime: new Date("2022-06-03"),
+          approximate_minimum_balance: "20",
+        },
+        {
+          tracked_token_account_id: updatedTokenAccounts.find(
+            (account) => account.address === "account3"
+          ).id,
+          datetime: new Date("2022-06-03"),
+          approximate_minimum_balance: "300",
+        },
+        {
+          tracked_token_account_id: updatedTokenAccounts.find(
+            (account) => account.address === "account4"
+          ).id,
+          datetime: new Date("2022-06-03"),
+          approximate_minimum_balance: "40",
+        },
+      ]);
+
+    // changes should only have the net new rows and not the unchanged ones
```

Comment:
My personal preference would be breaking out these comments into the text of new `it` test cases. That way you can see the human readable expectations during each test CLI run. 

That way it's not just 1 400 line long test.

Reply:
Haha yeah good call out, I'll refactor this into separate tests in another PR